### ISSUE TITLE
Cannot build for M1

### ISSUE THREAD

**Issue by MrBenFTW** [obsproject/obs-studio#4266]
Hi, I am trying to rebuild OBS so it can run natively on M1. This is a fresh system. I have installed home-brew, updated and cleaned it up.

I cloned OBS using the recursive flag to download the sub modules, then changed directory to it and then ran `./CI/full-build-macos.sh -b`

This is the output:

```
benmitchell@Scylla obs-studio % ./CI/full-build-macos.sh -b
  + Fetching OBS tags...
[OBS-Studio] Installing Homebrew dependencies
Using akeru-inc/tap
Using cmake
Using freetype
Using cmocka
Using akeru-inc/tap/xcnotary
Homebrew Bundle complete! 5 Brewfile dependencies now installed.
[OBS-Studio] Setting up pre-built macOS OBS dependencies v2020-12-22
  + Download...
######################################################################### 100.0%
  + Unpack...
[OBS-Studio] Setting up pre-built dependency QT v5.15.2
  + Download...
######################################################################### 100.0%
  + Unpack...
[OBS-Studio] Building dependency CEF v4183
  + Download...
######################################################################### 100.0%
  + Unpack...
  + Fix tests...
  + Run CMAKE...
CMake Warning (dev) at CMakeLists.txt:187 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable CEF_ROOT is set to:

    /Users/benmitchell/Files/obs-build-dependencies/cef_binary_4183_macosx64

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning at cmake/cef_variables.cmake:36 (message):
  No PROJECT_ARCH value specified, using x86_64
Call Stack (most recent call first):
  cmake/FindCEF.cmake:38 (include)
  CMakeLists.txt:187 (find_package)


CMake Warning at cmake/cef_variables.cmake:50 (message):
  No CMAKE_BUILD_TYPE value selected, using Release
Call Stack (most recent call first):
  cmake/FindCEF.cmake:38 (include)
  CMakeLists.txt:187 (find_package)


-- *** CEF CONFIGURATION SETTINGS ***
-- Generator:                    Unix Makefiles
-- Platform:                     Darwin
-- Project architecture:         x86_64
-- Build type:                   Release
-- Binary distribution root:     /Users/benmitchell/Files/obs-build-dependencies/cef_binary_4183_macosx64
-- Base SDK:                     /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-- Target SDK:                   10.13
-- CEF sandbox:                  ON
-- Standard libraries:           -lpthread;-framework Cocoa;-framework AppKit
-- Compile defines:              __STDC_CONSTANT_MACROS;__STDC_FORMAT_MACROS;CEF_USE_SANDBOX
-- Compile defines (Debug):      
-- Compile defines (Release):    
-- C compile flags:              -fno-strict-aliasing;-fstack-protector;-funwind-tables;-fvisibility=hidden;-Wall;-Werror;-Wextra;-Wendif-labels;-Wnewline-eof;-Wno-missing-field-initializers;-Wno-unused-parameter;-mmacosx-version-min=10.13 -std=c99
-- C compile flags (Debug):      -O0;-g 
-- C compile flags (Release):    -O3 
-- C++ compile flags:            -fno-strict-aliasing;-fstack-protector;-funwind-tables;-fvisibility=hidden;-Wall;-Werror;-Wextra;-Wendif-labels;-Wnewline-eof;-Wno-missing-field-initializers;-Wno-unused-parameter;-mmacosx-version-min=10.13 -fno-exceptions;-fno-rtti;-fno-threadsafe-statics;-fobjc-call-cxx-cdtors;-fvisibility-inlines-hidden;-std=gnu++11;-Wno-narrowing;-Wsign-compare;-Wno-undefined-var-template
-- C++ compile flags (Debug):    -O0;-g 
-- C++ compile flags (Release):  -O3 
-- Exe link flags:               -Wl,-search_paths_first;-Wl,-ObjC;-Wl,-pie 
-- Exe link flags (Debug):        
-- Exe link flags (Release):     -Wl,-dead_strip 
-- Shared link flags:            -Wl,-search_paths_first;-Wl,-ObjC;-Wl,-pie 
-- Shared link flags (Debug):     
-- Shared link flags (Release):  -Wl,-dead_strip 
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/benmitchell/Files/obs-build-dependencies/cef_binary_4183_macosx64/build
  + Build...
[100%] Built target libcef_dll_wrapper
[OBS-Studio] Setting up dependency VLC v3.0.8
  + Download...
######################################################################### 100.0%
  + Unpack ...
[OBS-Studio] Setting up dependency Sparkle v1.23.0 (might prompt for password)
  + Download...
######################################################################### 100.0%
  + Unpack...
  + Copy to destination...
 + Warning - Sparkle framework already found in /Library/Frameworks
 + CCache is not available.
[OBS-Studio] Run CMAKE for OBS...
-- OBS_VERSION: 26.1.2-200-g848395f32
-- No Native SSE2 SIMD Support - Using SIMDE
-- Found FFmpeg: /tmp/obsdeps/lib/../lib/libavcodec.dylib (found version "58.54.100") found components: avcodec avdevice avutil avformat 
-- Scripting: Luajit supported
-- Python support not found.
-- Using system Jansson library
-- Found FFmpeg: /tmp/obsdeps/lib/../lib/libavformat.dylib (found version "58.29.100") found components: avformat avutil swscale swresample avcodec 
-- Using libavcodec for image loading in libobs
-- Checking for module 'jack'
--   No package 'jack' found
-- JACK Audio Connection Kit not found, disabling JACK plugin
-- Looking for Chromium Embedded Framework in /Users/benmitchell/Files/obs-build-dependencies/cef_binary_4183_macosx64
-- Using the bundled VST header.
-- Libfdk not found - obs-libfdk plugin disabled
-- Found FFmpeg: /tmp/obsdeps/lib/../lib/libavcodec.dylib (found version "58.54.100") found components: avcodec avfilter avdevice avutil swscale avformat swresample 
-- Found FFmpeg: /tmp/obsdeps/lib/../lib/libavcodec.dylib (found version "58.54.100") found components: avcodec avutil avformat 
-- Checking for module 'libftl'
--   No package 'libftl' found
-- Found ftl-sdk: ftl outputs enabled
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/benmitchell/Files/obs-studio/build
[OBS-Studio] Build OBS...
[  1%] Built target glad
[  3%] Built target caption
[  3%] Automatic MOC for target obs-browser-page_renderer
[  4%] Built target mac-dal-plugin
[  5%] Automatic MOC for target obs-browser-page_gpu
[  5%] Built target obslua_swig_compilation
[  5%] Built target media-playback
[  9%] Built target syphon-framework
[  9%] Built target obs-browser-page_renderer_autogen
[  9%] Automatic MOC for target obs-browser-page
[  9%] Automatic MOC for target obs-browser-page_plugin
[  9%] Built target obs-browser-page_gpu_autogen
[  9%] Linking CXX executable "OBS Helper (Renderer).app/Contents/MacOS/OBS Helper (Renderer)"
[  9%] Built target obs-browser-page_autogen
[  9%] Built target obs-browser-page_plugin_autogen
[  9%] Linking C shared library libobs.dylib
[  9%] Linking CXX executable "OBS Helper (GPU).app/Contents/MacOS/OBS Helper (GPU)"
ld: warning: ignoring file /Users/benmitchell/Files/obs-build-dependencies/cef_binary_4183_macosx64/build/libcef_dll_wrapper/libcef_dll_wrapper.a, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
[ 10%] Linking CXX executable "OBS Helper (Plugin).app/Contents/MacOS/OBS Helper (Plugin)"
[ 10%] Linking CXX executable "OBS Helper.app/Contents/MacOS/OBS Helper"
ld: warning: ignoring file /tmp/obsdeps/lib/../lib/libavcodec.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /tmp/obsdeps/lib/../lib/libjansson.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /tmp/obsdeps/lib/../lib/libavutil.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /tmp/obsdeps/lib/../lib/libavformat.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /tmp/obsdeps/lib/../lib/libswscale.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /tmp/obsdeps/lib/../lib/libswresample.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /Users/benmitchell/Files/obs-build-dependencies/cef_binary_4183_macosx64/build/libcef_dll_wrapper/libcef_dll_wrapper.a, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "CefExecuteProcess(CefMainArgs const&, scoped_refptr<CefApp>, void*)", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefV8Value::CreateBool(bool)", referenced from:
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "CefV8Value::CreateObject(scoped_refptr<CefV8Accessor>, scoped_refptr<CefV8Interceptor>)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
  "CefV8Value::CreateString(CefStringBase<CefStringTraitsUTF16> const&)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
  "CefV8Value::CreateFunction(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Handler>)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
  "CefV8Context::GetCurrentContext()", referenced from:
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "CefProcessMessage::Create(CefStringBase<CefStringTraitsUTF16> const&)", referenced from:
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Valuld: warning: ignoring file /Users/benmitchell/Files/obs-build-dependencies/cef_binary_4183_macosx64/build/libcef_dll_wrapper/libcef_dll_wrapper.a, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
e>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "CefScopedLibraryLoader::Load(bool)", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefScopedLibraryLoader::CefScopedLibraryLoader()", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefScopedLibraryLoader::~CefScopedLibraryLoader()", referenced from:
      _main in obs-browser-page-main.cpp.o
  "_cef_string_utf16_clear", referenced from:
      BrowserApp::OnRegisterCustomSchemes(CefSchemeRegistrar*) in browser-app.cpp.o
      CefStringBase<CefStringTraitsUTF16>::~CefStringBase() in browser-app.cpp.o
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
      BrowserApp::ExecuteJSFunction(scoped_refptr<CefBrowser>, char const*, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > >) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      ...
  "_cef_string_utf16_cmp", referenced from:
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "_cef_string_utf16_to_utf8", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "_cef_string_utf8_clear", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "_cef_string_utf8_to_utf16", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      CefStringBase<CefStringTraitsUTF16>::CefStringBase(char const*) in browser-app.cpp.o
ld: symbol(s) not found for architecture arm64
Undefined symbols for architecture arm64:
  "CefExecuteProcess(CefMainArgs const&, scoped_refptr<CefApp>, void*)", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefV8Value::CreateBool(bool)", referenced from:
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "CefV8Value::CreateObject(scoped_refptr<CefV8Accessor>, scoped_refptr<CefV8Interceptor>)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
  "CefV8Value::CreateString(CefStringBase<CefStringTraitsUTF16> const&)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
ld: warning: ignoring file /Users/benmitchell/Files/obs-build-dependencies/cef_binary_4183_macosx64/build/libcef_dll_wrapper/libcef_dll_wrapper.a, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
  "CefV8Value::CreateFunction(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Handler>)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
  "CefV8Context::GetCurrenclang: error: linker command failed with exit code 1 (use -v to see invocation)
tContext()", referenced from:
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "CefProcessMessage::Create(CefStringBase<CefStringTraitsUTF16> const&)", referenced from:
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "CefScopedLibraryLoader::Load(bool)", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefScopedLibraryLoader::CefScopedLibraryLoader()", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefScopedLibraryLoader::~CefScopedLibraryLoader()", referenced from:
      _main in obs-browser-page-main.cpp.o
  "_cef_string_utf16_clear", referenced from:
      BrowserApp::OnRegisterCustomSchemes(CefSchemeRegistrar*) in browser-app.cpp.o
      CefStringBase<CefStringTraitsUTF16>::~CefStringBase() in browser-app.cpp.o
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
      BrowserApp::ExecuteJSFunction(scoped_refptr<CefBrowser>, char const*, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > >) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_promake[2]: *** [plugins/obs-browser/OBS Helper (Renderer).app/Contents/MacOS/OBS Helper (Renderer)] Error 1
make[2]: *** Archive member `plugins/obs-browser/OBS Helper (Renderer).app/Contents/MacOS/OBS Helper (Renderer)' may be bogus; not deleted
cess_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      ...
  "_cef_string_utf16_cmp", referenced from:
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "_cef_string_utf16_to_utf8", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
    make[1]: *** [plugins/obs-browser/CMakeFiles/obs-browser-page_renderer.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
  BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "_cef_string_utf8_clear", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "_cef_string_utf8_to_utf16", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      CefStringBase<CefStringTraitsUTF16>::CefStringBase(char const*) in browser-app.cpp.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [plugins/obs-browser/OBS Helper (GPU).app/Contents/MacOS/OBS Helper (GPU)] Error 1
make[2]: *** Archive member `plugins/obs-browser/OBS Helper (GPU).app/Contents/MacOS/OBS Helper (GPU)' may be bogus; not deleted
make[1]: *** [plugins/obs-browser/CMakeFiles/obs-browser-page_gpu.dir/all] Error 2
Undefined symbols for architecture arm64:
  "CefExecuteProcess(CefMainArgs const&, scoped_refptr<CefApp>, void*)", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefV8Value::CreateBool(bool)", referenced from:
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "CefV8Value::CreateObject(scoped_refptr<CefV8Accessor>, scoped_refptr<CefV8Interceptor>)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
  "CefV8Value::CreateString(CefStringBase<CefStringTraitsUTF16> const&)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
  "CefV8Value::CreateFunction(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Handler>)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
  "CefV8Context::GetCurrentContext()", referenced from:
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "CefProcessMessage::Create(CefStringBase<CefStringTraitsUTF16> const&)", referenced from:
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "CefScopedLibraryLoader::Load(bool)", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefScopedLibraryLoader::CefScopedLibraryLoader()", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefScopedLibraryLoader::~CefScopedLibraryLoader()", referenced from:
      _main in obs-browser-page-main.cpp.o
  "_cef_string_utf16_clear", referenced from:
      BrowserApp::OnRegisterCustomSchemes(CefSchemeRegistrar*) in browser-app.cpp.o
      CefStringBase<CefStringTraitsUTF16>::~CefStringBase() in browser-app.cpp.o
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
      BrowserApp::ExecuteJSFunction(scoped_refptr<CefBrowser>, char const*, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > >) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      ...
  "_cef_string_utf16_cmp", referenced from:
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "_cef_string_utf16_to_utf8", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "_cef_string_utf8_clear", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "_cef_string_utf8_to_utf16", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      CefStringBase<CefStringTraitsUTF16>::CefStringBase(char const*) in browser-app.cpp.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [plugins/obs-browser/OBS Helper (Plugin).app/Contents/MacOS/OBS Helper (Plugin)] Error 1
make[2]: *** Archive member `plugins/obs-browser/OBS Helper (Plugin).app/Contents/MacOS/OBS Helper (Plugin)' may be bogus; not deleted
make[1]: *** [plugins/obs-browser/CMakeFiles/obs-browser-page_plugin.dir/all] Error 2
Undefined symbols for architecture arm64:
  "CefExecuteProcess(CefMainArgs const&, scoped_refptr<CefApp>, void*)", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefV8Value::CreateBool(bool)", referenced from:
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "CefV8Value::CreateObject(scoped_refptr<CefV8Accessor>, scoped_refptr<CefV8Interceptor>)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
  "CefV8Value::CreateString(CefStringBase<CefStringTraitsUTF16> const&)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
  "CefV8Value::CreateFunction(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Handler>)", referenced from:
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
  "CefV8Context::GetCurrentContext()", referenced from:
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "CefProcessMessage::Create(CefStringBase<CefStringTraitsUTF16> const&)", referenced from:
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "CefScopedLibraryLoader::Load(bool)", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefScopedLibraryLoader::CefScopedLibraryLoader()", referenced from:
      _main in obs-browser-page-main.cpp.o
  "CefScopedLibraryLoader::~CefScopedLibraryLoader()", referenced from:
      _main in obs-browser-page-main.cpp.o
  "_cef_string_utf16_clear", referenced from:
      BrowserApp::OnRegisterCustomSchemes(CefSchemeRegistrar*) in browser-app.cpp.o
      CefStringBase<CefStringTraitsUTF16>::~CefStringBase() in browser-app.cpp.o
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnContextCreated(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, scoped_refptr<CefV8Context>) in browser-app.cpp.o
      BrowserApp::ExecuteJSFunction(scoped_refptr<CefBrowser>, char const*, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > >) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      ...
  "_cef_string_utf16_cmp", referenced from:
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      BrowserApp::Execute(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefV8Value>, std::__1::vector<scoped_refptr<CefV8Value>, std::__1::allocator<scoped_refptr<CefV8Value> > > const&, scoped_refptr<CefV8Value>&, CefStringBase<CefStringTraitsUTF16>&) in browser-app.cpp.o
  "_cef_string_utf16_to_utf8", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "_cef_string_utf8_clear", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
  "_cef_string_utf8_to_utf16", referenced from:
      BrowserApp::OnBeforeCommandLineProcessing(CefStringBase<CefStringTraitsUTF16> const&, scoped_refptr<CefCommandLine>) in browser-app.cpp.o
      BrowserApp::SetFrameDocumentVisibility(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, bool) in browser-app.cpp.o
      BrowserApp::OnProcessMessageReceived(scoped_refptr<CefBrowser>, scoped_refptr<CefFrame>, cef_process_id_t, scoped_refptr<CefProcessMessage>) in browser-app.cpp.o
      CefStringBase<CefStringTraitsUTF16>::CefStringBase(char const*) in browser-app.cpp.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [plugins/obs-browser/OBS Helper.app/Contents/MacOS/OBS Helper] Error 1
make[1]: *** [plugins/obs-browser/CMakeFiles/obs-browser-page.dir/all] Error 2
Undefined symbols for architecture arm64:
  "_av_dict_copy", referenced from:
      _media_remux_job_create in media-remux.c.o
  "_av_find_best_stream", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_av_frame_alloc", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_av_frame_free", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_av_freep", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
      _audio_resampler_create in audio-resampler-ffmpeg.c.o
      _audio_resampler_destroy in audio-resampler-ffmpeg.c.o
      _audio_resampler_resample in audio-resampler-ffmpeg.c.o
      _video_scaler_create in video-scaler-ffmpeg.c.o
      _video_scaler_destroy in video-scaler-ffmpeg.c.o
  "_av_image_alloc", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
      _video_scaler_create in video-scaler-ffmpeg.c.o
  "_av_interleaved_write_frame", referenced from:
      _media_remux_job_process in media-remux.c.o
  "_av_packet_unref", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
      _media_remux_job_process in media-remux.c.o
  "_av_pix_fmt_desc_get", referenced from:
      _video_scaler_create in video-scaler-ffmpeg.c.o
  "_av_read_frame", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
      _media_remux_job_process in media-remux.c.o
  "_av_rescale_q", referenced from:
      _media_remux_job_process in media-remux.c.o
  "_av_rescale_q_rnd", referenced from:
      _media_remux_job_process in media-remux.c.o
  "_av_rescale_rnd", referenced from:
      _audio_resampler_resample in audio-resampler-ffmpeg.c.o
  "_av_samples_alloc", referenced from:
      _audio_resampler_resample in audio-resampler-ffmpeg.c.o
  "_av_strerror", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
      _media_remux_job_process in media-remux.c.o
  "_av_write_trailer", referenced from:
      _media_remux_job_process in media-remux.c.o
  "_avcodec_alloc_context3", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_avcodec_find_decoder", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_avcodec_free_context", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_avcodec_open2", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_avcodec_parameters_copy", referenced from:
      _media_remux_job_create in media-remux.c.o
  "_avcodec_parameters_to_context", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_avcodec_receive_frame", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_avcodec_send_packet", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_avformat_alloc_output_context2", referenced from:
      _media_remux_job_create in media-remux.c.o
  "_avformat_close_input", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
      _media_remux_job_create in media-remux.c.o
      _media_remux_job_destroy in media-remux.c.o
  "_avformat_find_stream_info", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
      _media_remux_job_create in media-remux.c.o
  "_avformat_free_context", referenced from:
      _media_remux_job_create in media-remux.c.o
      _media_remux_job_destroy in media-remux.c.o
  "_avformat_new_stream", referenced from:
      _media_remux_job_create in media-remux.c.o
  "_avformat_open_input", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
      _media_remux_job_create in media-remux.c.o
  "_avformat_write_header", referenced from:
      _media_remux_job_process in media-remux.c.o
  "_avio_close", referenced from:
      _media_remux_job_create in media-remux.c.o
      _media_remux_job_destroy in media-remux.c.o
  "_avio_open", referenced from:
      _media_remux_job_create in media-remux.c.o
  "_json_array", referenced from:
      _obs_data_to_json in obs-data.c.o
  "_json_array_append_new", referenced from:
      _obs_data_to_json in obs-data.c.o
  "_json_array_get", referenced from:
      _obs_data_add_json_item in obs-data.c.o
  "_json_array_size", referenced from:
      _obs_data_add_json_item in obs-data.c.o
  "_json_delete", referenced from:
      _obs_data_create_from_json in obs-data.c.o
      _obs_data_get_json in obs-data.c.o
      _obs_data_save_json in obs-data.c.o
      _obs_data_save_json_safe in obs-data.c.o
  "_json_dumps", referenced from:
      _obs_data_get_json in obs-data.c.o
      _obs_data_save_json in obs-data.c.o
      _obs_data_save_json_safe in obs-data.c.o
  "_json_false", referenced from:
      _obs_data_to_json in obs-data.c.o
  "_json_integer", referenced from:
      _obs_data_to_json in obs-data.c.o
  "_json_integer_value", referenced from:
      _obs_data_add_json_item in obs-data.c.o
  "_json_loads", referenced from:
      _obs_data_create_from_json in obs-data.c.o
  "_json_object", referenced from:
      _obs_data_to_json in obs-data.c.o
  "_json_object_iter", referenced from:
      _obs_data_create_from_json in obs-data.c.o
      _obs_data_add_json_item in obs-data.c.o
  "_json_object_iter_key", referenced from:
      _obs_data_create_from_json in obs-data.c.o
      _obs_data_add_json_item in obs-data.c.o
  "_json_object_iter_next", referenced from:
      _obs_data_create_from_json in obs-data.c.o
      _obs_data_add_json_item in obs-data.c.o
  "_json_object_iter_value", referenced from:
      _obs_data_create_from_json in obs-data.c.o
      _obs_data_add_json_item in obs-data.c.o
  "_json_object_key_to_iter", referenced from:
      _obs_data_create_from_json in obs-data.c.o
      _obs_data_add_json_item in obs-data.c.o
  "_json_object_set_new", referenced from:
      _obs_data_to_json in obs-data.c.o
  "_json_real", referenced from:
      _obs_data_to_json in obs-data.c.o
  "_json_real_value", referenced from:
      _obs_data_add_json_item in obs-data.c.o
  "_json_string", referenced from:
      _obs_data_to_json in obs-data.c.o
  "_json_string_value", referenced from:
      _obs_data_add_json_item in obs-data.c.o
  "_json_true", referenced from:
      _obs_data_to_json in obs-data.c.o
  "_swr_alloc_set_opts", referenced from:
      _audio_resampler_create in audio-resampler-ffmpeg.c.o
  "_swr_convert", referenced from:
      _audio_resampler_resample in audio-resampler-ffmpeg.c.o
  "_swr_free", referenced from:
      _audio_resampler_create in audio-resampler-ffmpeg.c.o
      _audio_resampler_destroy in audio-resampler-ffmpeg.c.o
  "_swr_get_delay", referenced from:
      _audio_resampler_resample in audio-resampler-ffmpeg.c.o
  "_swr_init", referenced from:
      _audio_resampler_create in audio-resampler-ffmpeg.c.o
  "_swr_set_matrix", referenced from:
      _audio_resampler_create in audio-resampler-ffmpeg.c.o
  "_sws_freeContext", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
      _video_scaler_create in video-scaler-ffmpeg.c.o
      _video_scaler_destroy in video-scaler-ffmpeg.c.o
  "_sws_getCachedContext", referenced from:
      _video_scaler_create in video-scaler-ffmpeg.c.o
  "_sws_getCoefficients", referenced from:
      _video_scaler_create in video-scaler-ffmpeg.c.o
  "_sws_getContext", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
  "_sws_scale", referenced from:
      _gs_create_texture_file_data in graphics-ffmpeg.c.o
      _video_scaler_scale in video-scaler-ffmpeg.c.o
  "_sws_setColorspaceDetails", referenced from:
      _video_scaler_create in video-scaler-ffmpeg.c.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libobs/libobs.0.dylib] Error 1
make[1]: *** [libobs/CMakeFiles/libobs.dir/all] Error 2
make: *** [all] Error 2
  + ERROR during build step: cmake
benmitchell@Scylla obs-studio % 
```

I can't help but notice at the start it mentions x86_64. `-- Project architecture:         x86_64` maybe this is the issue?

My objective is to build the OBS project for M1/ARM64e

**Comment by WizardCM** [comment]
None of the dependencies have been built for native M1, so the official build script is not designed for it yet.

You can find some work at #4105 to adapt the OBS functions to work natively, and most of our dependencies are built at the [obs-deps](https://github.com/obsproject/obs-deps) repository.

**Comment by MrBenFTW** [comment]
> None of the dependencies have been built for native M1, so the official build script is not designed for it yet.
> 
> You can find some work at #4105 to adapt the OBS functions to work natively, and most of our dependencies are built at the [obs-deps](https://github.com/obsproject/obs-deps) repository.

Okay cool, if someone wants to tell me how to rebuild them for M1 I can go ahead and do so. Home-brew recently added full native support, you no longer need to run home-brew through rosetta

**Comment by kilinbox** [comment]
I'm testing the OBS by performing the following steps.

step1
Build obsdeps for arm64 with reference to "CI: Build OBS dependencies for Apple Silicon Macs #49".
https://github.com/obsproject/obs-deps/pull/49

step2
Some code changes based on "Bug fixes for Apple Silicon #4105".
https://github.com/obsproject/obs-studio/pull/4105

step3
Change full-build-macos.sh to use the obsdeps you built in "step1".

https://imgur.com/a/Y7UPehu

Thank you to all development in this work.

**Comment by kilinbox** [comment]
I'm testing the OBS by performing the following steps.

step1
Build obsdeps for arm64 with reference to "CI: Build OBS dependencies for Apple Silicon Macs #49".
obsproject/obs-deps#49

step2
Build the CEF.
https://bitbucket.org/chromiumembedded/cef-project/src/master/

step3
Some code changes based on "Bug fixes for Apple Silicon #4105".
#4105

step4
Change full-build-macos.sh to use the obsdeps built in "step1" and CEF built in "step2".

https://imgur.com/a/Y7UPehu

Thank you to all development in this work.

**Comment by PatTheMav** [comment]
Thank you all for working on getting OBS to run on macOS - in the interest of our collective sanity, we'd like to keep Github issues focused on actual "issues". OBS not building on M1 is not so much an issue but a current choice.

You are more than welcome to collaborate and participate in solving this in our Discord's development channel and also open PRs for this.

# Requirements

Roughly speaking, the necessary pieces for M1 support include (but are not limited to):

- Building all "small" macOS dependencies for x86 and ARM64 (ffmpeg and its statically linked dependencies, SWIG, libjansson, LuaJIT, freetype, speexDSP, rnnoise)
    - Building these for ARM64 might be easy in some cases, difficult in others
    - Open question is how to properly package the dynamic libraries, Apple suggests using XCFrameworks instead of "phat" libraries
- Building all "big" macOS dependencies for x86 and ARM64 (CEF and Qt)
    - Depends on proper ARM64 support upstream - CEF is working on it, but we'd also need to apply several semi-official patches to CEF for our purposes (compatibility unknown)
    - Qt support for ARM64 is "work in progress" with Qt 6.2 being the earliest possible version with support for it
- Packaging those dependencies
    - See XCFramework remark above.
- Building OBS on macOS for ARM64
    - As mentioned there is already a PR for it, but this work would realistically require the updated dependencies first 
- Creating an universal app bundle for macOS
    - Probably the least of all issues

The biggest blockers are CEF and (obviously) Qt, which introduces the additional question of whether or not to move to Qt6.

# Roadmap

Given M1's great x86 performance I'd really like not to rush this (OBS is a complex piece of software) and instead do it right:

1. Update dependencies to "known good" versions supporting M1 already
2. Update Qt to version supporting M1
3. Update CEF to version supporting M1
4. Fix all issues surrounding updated dependencies in OBS (x86)
5. Update CI to also build&package dependencies with M1 support
6. Update CI to build OBS for M1 with appropriate dependencies
7. Fail 😬
8. Fix bugs
9. Fail better 😁

The order of 1-3 might change, but given that with support for the new arch we also introduce changes of a new version there will be issues on the x86 side that need fixing as well (especially to rule out the architecture as the root cause).

PS: We do _not_ use Homebrew for our bundled dependencies. Homebrew is amazing (and we do use it to install build dependencies) but we need to use specific versions of dependencies and also include support for a wider range of macOS versions (e.g. Homebrew has deprecated 10.13 already, while we still support it), so we cannot simply use Homebrew bottles to build OBS, especially not for distribution.

**Comment by mathieunicolas** [comment]
> I'm testing the OBS by performing the following steps.
> 
> step1
> Build obsdeps for arm64 with reference to "CI: Build OBS dependencies for Apple Silicon Macs #49".
> [obsproject/obs-deps#49](https://github.com/obsproject/obs-deps/pull/49)
> 
> step2
> Build the CEF.
> https://bitbucket.org/chromiumembedded/cef-project/src/master/
> 
> step3
> Some code changes based on "Bug fixes for Apple Silicon #4105".
> #4105
> 
> step4
> Change full-build-macos.sh to use the obsdeps built in "step1" and CEF built in "step2".
> 
> https://imgur.com/a/Y7UPehu
> 
> Thank you to all development in this work.

Hi @kilinbox, and thanks for these steps. I'm not very experienced in what you describe here, but I really would like to compile OBS to a native Apple Silicon App, but I don't know how to do it. Would you please take the time to explain a bit more how to complete these steps ? Or, maybe since you posted this message, the steps have evolved ?
Need some help because I'm quite lost right now :p (I did succeed to build the CEF for example, but I don't know what to do with which files)
Thanks !!

**Comment by RytoEX** [comment]
My apologies for locking this issue.  We appreciate your enthusiasm and desire to see OBS with native M1 support, but per @PatTheMav in [his comment above](https://github.com/obsproject/obs-studio/issues/4266#issuecomment-783383104):
> Thank you all for working on getting OBS to run on macOS - in the interest of our collective sanity, we'd like to keep Github issues focused on actual "issues". OBS not building on M1 is not so much an issue but a current choice.
> 
> You are more than welcome to collaborate and participate in solving this in our Discord's development channel and also open PRs for this.

As mentioned by @PatTheMav, we have plans in place to reach an OBS release with native M1 support.  Some of the tasks may require more code or CI changes, and we welcome pull requests to help solve those issues.  Some of them require changes to our dependencies that we're waiting on, such as Qt and CEF, and we encourage efforts to help those dependencies support M1.

To my knowledge, Chromium and CEF have made progress towards native ARM64/M1 builds.  Both Chromium and CEF can be cross-compiled for ARM64/M1 from Intel hosts.  It is also now barely possible to build Chromium _on_ ARM64/M1 Macs, though the Chromium team has reservations and open issues they are tracking.
https://bitbucket.org/chromiumembedded/cef/issues/2981/macos-support-on-arm64
https://chromium.googlesource.com/chromium/src/+/a1acc7d53525abf2ff7aad231ac3159a2ee392cb/docs/mac_arm64.md

According to Qt's current timeline, they do not expect to officially support M1 until Qt 6.2, which is planned to be released September 2021.  At that point in time, we need to make sure OBS can build on Qt 6.2 and decide if we want to move all platforms to it, or still ship with Qt 5.15.2 on certain platforms (e.g., Ubuntu 18.04).  We've already done a lot of work for Qt6 compatibility and testing compiling OBS with Qt 6.0.x and the 6.1.0 betas, so we've got a little head start there at least.
https://bugreports.qt.io/browse/QTBUG-85279
https://bugreports.qt.io/browse/QTBUG-85546
https://wiki.qt.io/Qt_6.2_Release

It may be a while yet before we have an official OBS release with native M1 support, but please rest assured that we are working towards that goal.  Thank you all for your patience and understanding.
